### PR TITLE
Feat(docker): Distroless, Rootless release image & Buildx

### PR DESCRIPTION
**Pull request by mrzigha**
- Remove Alpine for runner container, switch to distroless.
- Switch to Docker buildx for building image.

## Summary by CodeRabbit

- **New Features**
  - Introduced a multi-stage Docker build for development, enhancing the build process and caching mechanisms.

- **Improvements**
  - Updated Dockerfile to use `scratch` for a more secure and efficient base image.
  - Switched to `docker buildx build` for optimized Docker builds.
  - Enhanced Google Cloud authentication and Docker configuration in CI pipelines.

- **Refactor**
  - Updated Docker image references from `beacond:kurtosis-local` to `beacond-dev:kurtosis-local` for better development environment consistency.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -19,12 +19,15 @@
 #######################################################
 
 ARG GO_VERSION=1.22.4
-ARG RUNNER_IMAGE=alpine:3.19
+ARG RUNNER_IMAGE=scratch
```
[the remark this answers is not in the document]
Comment (mrzigha) [reply Dockerfile:22]: There is no tag for [scratch](https://hub.docker.com/_/scratch/tags?page=&page_size=&ordering=-last_updated&name=) image.

File: Dockerfile [hunk 2]
```diff
@@ -134,17 +137,37 @@ RUN --mount=type=cache,target=/root/.cache/go-build \
...
+#######################################################
+
+FROM golang:${GO_VERSION}-alpine3.19 as adduser
+
+ARG FINAL_USERNAME
+ARG FINAL_UID
+ARG FINAL_GID
+
+RUN addgroup -g ${FINAL_GID} -S ${FINAL_USERNAME} && \
+    adduser -u ${FINAL_UID} -S -G ${FINAL_USERNAME} ${FINAL_USERNAME}
```
[the remark this answers is not in the document]
Comment (mrzigha) [reply Dockerfile]: Fixed !

**Comment by itsdevbear** [comment]
@mrzigha thanks for the PR! can you try running make start-devnet for me not sure if the container will run this way.

**Comment by chuck-bear** [comment]
This might be a breaking for our internal usage.

**Comment by mrzigha** [comment]
@itsdevbear @chuck-bear  Apparently your Kurtosis configuration doesn't support distroless. So I'm going to remove this feature from my PR.

**Comment by itsdevbear** [comment]
@mrzigha sounds good, there is probably value in having both types build 

1. alpine (or maybe there is something smaller) that is good to go
2. scratch, 

beacond:main-lean 

or something

**Comment by itsdevbear** [comment]
will merge buildx assume CI passes tho!

File: Dockerfile [hunk 3]
```diff
@@ -137,14 +137,11 @@ RUN --mount=type=cache,target=/root/.cache/go-build \
 ###        Stage 3 - Prepare the Final Image        ###
 #######################################################
 
-FROM ${RUNNER_IMAGE}
-
-# Build args
-ARG APP_NAME
+FROM alpine:3.14
```
Comment (itsdevbear) [inline Dockerfile:140]: revert back to ${{RUNNER_IMAGE}} and latest alpine

**Review by itsdevbear — CHANGES_REQUESTED** [review]

**Comment by mrzigha** [comment]
> @mrzigha sounds good, there is probably value in having both types build
> 
>     1. alpine (or maybe there is something smaller) that is good to go
> 
>     2. scratch,
> 
> 
> beacond:main-lean
> 
> or something

My original image worked in a prod-ready environment, the problem is that it's not compatible with Kurtosis, so I'm going to push your "devnet" (compatible) image and my image intended for a production environment if that suits you. I'll also apply the change requests you've made.

File: Dockerfile (hunk 3 above)
Comment (mrzigha) [reply Dockerfile]: Done

**Comment by itsdevbear** [comment]
closing for now

**Pull request closed without merging**